### PR TITLE
rust-crypto dependency removal

### PR DESCRIPTION
**Pull request by cheme**
This PR superseed #14 , is part of #80 split. Related to #84 .

It covers 'rust-crypto' crate removal.

'RustCrypto' crate is used :
- for some hashing function (not really a new dependency for parity eth)
- aes ecb
- aes ctr
The change to ctr is probably the more impacting for parity-ethereum.

I could also adopt a conservative approach and restore old unmaintained AES rust-crypto implementation and put RustCrypto in an alternative implementation crate.

Also worth remembering that we can enable aes-ni while building with RUSTFLAGS="-C target-feature=+aes,+sse2,+ssse3", for improved performances :
```
aes_ctr/10000           time:   [3.5767 us 3.5778 us 3.5789 us]                           
                        change: [-98.174% -98.174% -98.173%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  3 (3.00%) high mild
aes_ctr/100000          time:   [36.707 us 36.740 us 36.778 us]                            
                        change: [-98.091% -98.089% -98.087%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 14 outliers among 100 measurements (14.00%)
```

### CODE REVIEW

File: parity-crypto/src/hmac.rs [hunk 1]
```diff
@@ -86,3 +86,51 @@ impl VerifyKey<digest::Sha512> {
...
 	hmac::verify(&k.0, data, sig).is_ok()
 }
+
+
+#[cfg(test)]
+::tests_hmac!();
+
+#[cfg(test)]
+#[macro_export]
+macro_rules! tests_hmac {
```
Comment (sorpaas) [inline parity-crypto/src/hmac.rs]: It looks like we don't need this macro now?

Also by the way, you can create a separate test file and use `mod` with `path` attribute to include it multiple times. That should make everyone's formatter much happier.

Comment (cheme) [reply parity-crypto/src/hmac.rs]: Thanks for the 'path' siggestion, it will so much cleaner.

**Comment by tomaka** [comment]
Could we revive this PR? It probably needs some master merging.

**Comment by cheme** [comment]
no problem, will be updating that this afternoon :+1:

File: parity-crypto/src/aes.rs [hunk 2]
```diff
@@ -14,40 +14,195 @@
...
+
+	/// New encoder/decoder, no iv for ecb
+	pub fn new(key: &[u8]) -> Result<Self, SymmError> {
+		Ok(AesEcb256(Ecb::new_var(key, &[])?))
+	}
+
+	/// Encrypt data in place without padding. The data length must be a multiple
+	/// of the block size.
+	pub fn encrypt(self, content: &mut [u8]) -> Result<(), SymmError> {
+    let len = content.len();
```
Comment (tomaka) [inline parity-crypto/src/aes.rs]:
```suggestion
        let len = content.len();
```

**Comment by tomaka** [comment]
I don't know if anyone is watching this PR.
We have two positive reviews, so if anyone wants to review this as well, please express yourself here, otherwise I suggest we merge this in 24 hours.

**Pull request merged**